### PR TITLE
Feature nmd 218

### PR DESCRIPTION
**Pull request by Finka95**
Add some business logic in controllers

### CODE REVIEW

File: TicketSystem/TicketSystem/Controllers/MessageController.cs [hunk 1]
```diff
@@ -6,24 +6,23 @@
...
+        public MessageController()
         {
             _context = new ApplicationContext();
-            _logger = logger;
         }
 
-        [HttpPost(Name = "PostMessage")]
-        public Message Post(Message postMessage)
+        [HttpPost]
+        public async Task<IActionResult> Post(Message message)
```
Comment (CrazyKamenwik) [inline TicketSystem/TicketSystem/Controllers/MessageController.cs]: There's no need to use IActionResult as a return type. Use viewModel\model instead.

File: TicketSystem/TicketSystem/Controllers/MessageController.cs [hunk 2]
```diff
@@ -6,24 +6,23 @@
...
     [ApiController]
-    [Route("[controller]")]
+    [Route("api/[controller]")]
     public class MessageController : ControllerBase
     {
-        private readonly ILogger<MessageController> _logger;
         private readonly ApplicationContext _context;
 
-        public MessageController(ILogger<MessageController> logger)
+        public MessageController()
```
Comment (CrazyKamenwik) [inline TicketSystem/TicketSystem/Controllers/MessageController.cs]: Why did you delete logger?)

**Review by CrazyKamenwik — CHANGES_REQUESTED** [review]

File: TicketSystem/TicketSystem/Controllers/MessageController.cs (hunk 2 above)
Comment (v12code) [reply TicketSystem/TicketSystem/Controllers/MessageController.cs]: I don’t use it and deleted it)

File: TicketSystem/TicketSystem/Controllers/MessageController.cs (hunk 1 above)
Comment (v12code) [reply TicketSystem/TicketSystem/Controllers/MessageController.cs]: I can return Ok(message) in this method, is it good?

File: TicketSystem/TicketSystem/Controllers/TicketController.cs [hunk 3]
```diff
@@ -1,26 +1,36 @@
...
             _context = new ApplicationContext();
+            _logger = logger;
+        }
+
+        [HttpGet]
+        public async Task<IEnumerable<Ticket>> Get()
+        {
+            _logger.LogDebug("Get all tickets");
+            var tickets = await _context.Tickets.ToListAsync();
+            return tickets;
```
Comment (5Ka-me) [inline TicketSystem/TicketSystem/Controllers/TicketController.cs:26]: Add empty line

File: TicketSystem/TicketSystem/Controllers/TicketController.cs [hunk 4]
```diff
@@ -1,26 +1,36 @@
...
         }
 
-        [HttpGet(Name = "GetTickets")]
-        public IEnumerable<Ticket> Get()
+        [HttpGet("{id}")]
+        public async Task<Ticket?> Get(int id)
         {
-             return _context.Tickets.ToList();
+            _logger.LogDebug("Find ticket by id {id}", id);
+            return await _context.Tickets.FindAsync(id);
```
Comment (5Ka-me) [inline TicketSystem/TicketSystem/Controllers/TicketController.cs:33]: Add empty line

File: TicketSystem/TicketSystem/Controllers/UserController.cs [hunk 5]
```diff
@@ -1,26 +1,76 @@
...
+        public async Task<IEnumerable<User>> Get()
+        {
+            return await _context.Users.ToListAsync();
+        }
+
+        [HttpGet("{id}")]
+        public async Task<User?> Get(int id)
+        {
+            _logger.LogDebug("Find user by id {id}", id);
+            return await _context.Users.FindAsync(id);
```
Comment (5Ka-me) [inline TicketSystem/TicketSystem/Controllers/UserController.cs:32]: Add empty line

**Review by 5Ka-me — APPROVED** [review]

**Review by CrazyKamenwik — APPROVED** [review]

**Review by blackberts — APPROVED** [review]
